src/main/java/org/testng/internal/ParameterInfo.java

```diff
@@ -2,18 +2,30 @@
 
 public class ParameterInfo implements IParameterInfo {
   private Object instance;
+  private int index;
```

Comment:
Could you put it `final`?

Reply:
If I will set `final` to `private int index;` I should delete the following constructor and its use:
```
  public ParameterInfo(Object instance, Object[] parameters) {
    this.instance = instance;
    this.parameters = parameters;
  }
```

Reply:
That's exactly my point: when this constructor is used, `index` is initialized to `0` by default and I'm not sure it is the expected behavior.
Instead, you can init `index` yourself and explain why into a comment (maybe `-1` is a better default value) or remove the first constructor as you propose.

Reply:
Thank you @juherr, I have updated it.